js/app/historyStore.js

```diff
@@ -91,9 +92,11 @@ const HistoryStore = new Lang.Class({
         Dispatcher.get_default().register((payload) => {
             switch(payload.action_type) {
                 case Actions.HISTORY_BACK_CLICKED:
+                    this.get_previous_item().entry_point = EntryPoints.NAV_BUTTON_CLICKED;
```

Comment:
But I don't think you can have two open views of the same item at the same time, so just the key variant should be sufficient to deduplicate them. So really you only need the entry point info at the time you record the start metric, which is also exactly when you know the entry point info; so you don't need to save it anywhere. I still think it should be possible.